hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbsoluteResourceCapacityCalculator.java

```diff
@@ -59,7 +65,20 @@ public double calculateMaximumResource(
   public void updateCapacitiesAfterCalculation(
       ResourceCalculationDriver resourceCalculationDriver, CSQueue queue, String label) {
     CapacitySchedulerQueueCapacityHandler.setQueueCapacities(
-        resourceCalculationDriver.getUpdateContext().getUpdatedClusterResource(label), queue, label);
+        resourceCalculationDriver.getUpdateContext()
+            .getUpdatedClusterResource(label), queue, label);
+
+    if (queue.getParent() != null) {
```

Comment:
check: is this still needed?

Reply:
Looks like it's no longer needed, I'll run 2x test with both legacy and the new mode, without this change.